crates/iroha_core/src/smartcontracts/isi/triggers/mod.rs

```diff
@@ -66,10 +67,9 @@ pub mod isi {
                         match latest_block_time {
                             // Genesis block
                             None => {
-                                let genesis_creation_time_ms = genesis_creation_time_ms
-                                    .expect("INTERNAL BUG: genesis creation time not set");
-
-                                if schedule.start_ms < genesis_creation_time_ms {
+                                if Duration::from_millis(schedule.start_ms)
+                                    < state_transaction.curr_block.creation_time()
+                                {
```

Comment:
`match latest_block_time { ... }` can be simplified:

```rust
                        let latest_block_time = latest_block_time.unwrap_or_else(|| {
                            // Genesis block
                            state_transaction.curr_block.creation_time()
                        });
                        if schedule.start() < latest_block_time {
                            return Err(Error::InvalidParameter(
                                InvalidParameterError::TimeTriggerInThePast,
                            ));
                        }
```

Reply:
done